bndtools.m2e/src/bndtools/m2e/MavenWorkspaceRepository.java

```diff
@@ -177,25 +177,33 @@ public boolean isEmpty() {
 
 	@Override
 	public void mavenProjectChanged(MavenProjectChangedEvent[] events, IProgressMonitor monitor) {
+		boolean changed = false;
+
 		for (MavenProjectChangedEvent event : ((events != null) ? events : new MavenProjectChangedEvent[0])) {
 			Kind kind = Kind.get(event);
 			logger.debug("{}: mavenProjectChanged({}, {})", getName(), kind, event.getSource());
 			switch (kind) {
 				case ADDED :
 				case CHANGED :
 					IMavenProjectFacade projectFacade = event.getMavenProject();
-					IProject iProject = projectFacade.getProject();
-					index(iProject, collect(projectFacade, monitor));
+					Set<File> collected = collect(projectFacade, monitor);
+					if (!collected.isEmpty()) {
+						index(projectFacade.getProject(), collected);
+						changed = true;
+					}
 					break;
 				case REMOVED :
-					remove(event.getOldMavenProject()
+					changed = remove(event
```

Comment:
Since you are in a loop here, you probably do not want to potentially set changed from true back to false. So you want something like:

```java
if (remove(event.getOldMavenProject().getProject())) {
  changed = true;
}
```

Reply:
Or `changed |= remove(...`.

Reply:
I noticed this and was already fixing it :) I actually needed to fix this exact issue in a wider scope.